it/uk/gov/hmrc/nrs/retrieval/NrsRetrievalIntegrationSpec.scala

```diff
@@ -0,0 +1,217 @@
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.{BeforeAndAfterEach, Matchers, WordSpecLike}
+import org.scalatestplus.play.guice.GuiceOneServerPerSuite
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.ws.WSClient
+
+class NrsRetrievalIntegrationSpec
+  extends WordSpecLike
+    with Matchers
+    with ScalaFutures
+    with GuiceOneServerPerSuite
+    with WireMockSupport
+    with IntegrationPatience
+    with BeforeAndAfterEach {
+  private val xApiKeyHeader = "X-API-Key"
+  private val xApiKey = "xApiKey"
+  private val equalsXApiKey = new EqualToPattern(xApiKey)
+
+  override lazy val port: Int = 19391
+
+  private lazy val local = s"http://localhost:$port"
+  private lazy val serviceRoot = s"$local/nrs-retrieval"
+
+  /*
+   * Use WSClient rather than HttpClient here because:
+   * 1. we have implemented our own HttpClient.HEAD (with opinionated behaviour) and we don't want to mix tests of
+   *    this with tests of service endpoint behaviour.
+   * 2. when we update http-verbs the tests are not subtly changed.
+   */
+  private lazy val wsClient = fakeApplication().injector.instanceOf[WSClient]
+
+  override def fakeApplication(): Application =
+    new GuiceApplicationBuilder().configure(Map[String, Any](
+      "awsservices.nonrepSubmissionPing.url" -> wireMockBaseUrl,
+      "awsservices.nonrepRetrievalPing.url" -> wireMockBaseUrl,
+      "awsservices.nonrepRetrieval.url" -> wireMockBaseUrl,
+      "auditing.enabled" -> false,
+      "metrics.jvm" -> false)
+    ).build()
+
+  override def beforeEach(): Unit = WireMock.reset()
+
+  private def wsClientWithXApiKeyHeader(url: String) =
+    wsClient.url(url).withHttpHeaders((xApiKeyHeader, xApiKey))
+
+  private def mappingBuilder(headMapping: MappingBuilder, withHeaders: Boolean) =
+    if (withHeaders) {
+      headMapping.withHeader(xApiKeyHeader, equalsXApiKey)
+    } else {
+      headMapping
+    }
+
+  "GET /ping/ping" should {
+    "succeed" in {
+      wsClient.url(s"$local/ping/ping").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "GET /nrs-retrieval/submission/ping" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/submission/ping").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "GET /nrs-retrieval/retrieval/ping" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/retrieval/ping").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "GET /nrs-retrieval/submission-metadata" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/submission-metadata").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "GET /nrs-retrieval/submission-bundles/:vaultId/:archiveId" should {
+    "succeed" in {
+      wsClient.url(s"$serviceRoot/submission-bundles/vaultId/archiveId").get().futureValue.status shouldBe OK
+    }
+  }
+
+  "HEAD /nrs-retrieval/submission-bundles/:vaultId/:archiveId" should {
```

Comment:
this endpoint and the next require headers and also rewrite response codes, so are tested in more detail

Reply:
are /nrs-retrieval/submission-metadata and /nrs-retrieval/submission-bundles/:vaultId/:archiveId endpoints accepting GET requests any different?

Reply:
Maybe.

The `http-verbs` implementation may well have changed, so it might be that there's a difference with `400` and `404` response handling with the `GET` implementation.

`GET /nrs-retrieval/submission-metadata` and
`GET  /nrs-retrieval/submission-bundles/:vaultId/:archiveId`

don't do any explicit response code re-writing or set any request headers so I felt it might be overkill to add extra it:tests. 

I could do that though?

Reply:
to be clear the only difference I've seen is that if http-verbs encounters a 400 from upstream, our endpoint will now return a 500 rather than a 400.

Not sure this maters - the frontend doesn't have any explicit handling for those return codes.

Reply:
> I could do that though?

hard to say, all of these endpoint must provide x-api-key to be allowed to communicate with the backend service, so at least that one entry should be added to request headers